src/datumaro/experimental/categories.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (C) 2025 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+"""
+Categories definitions for the experimental dataset system.
+
+This module provides category management functionality using standard dataclasses
+instead of attrs, taking inspiration from the original Categories implementation.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from enum import IntEnum
+from typing import Dict, Iterator, List, NamedTuple, Optional, Tuple, Union
+
+
+class GroupType(IntEnum):
+    """Types of label groups for organizing labels."""
+
+    EXCLUSIVE = 0
+    INCLUSIVE = 1
+    RESTRICTED = 2
```

Comment:
What is a restricted group type? Would be good to have a description for these options.